libqtile/backend/wayland/xdgwindow.py

```diff
@@ -194,25 +194,36 @@ def handle_activation_request(self, focus_on_window_activation: str) -> None:
             # Likely still pending, ignore this request.
             return
 
+        # For "focus", always focus the window.
         if focus_on_window_activation == "focus":
-            logger.debug("Focusing window (focus_on_window_activation='focus')")
+            logger.debug(
+                "Focusing window (focus_on_window_activation='%s')", focus_on_window_activation
+            )
             self.qtile.current_screen.set_group(self.group)
             self.group.focus(self)
+            return
 
-        elif focus_on_window_activation == "smart":
-            if not self.group.screen:
-                logger.debug("Ignoring focus request (focus_on_window_activation='smart')")
-            elif self.group.screen == self.qtile.current_screen:
-                logger.debug("Focusing window (focus_on_window_activation='smart')")
-                self.qtile.current_screen.set_group(self.group)
-                self.group.focus(self)
-            else:
-                self._urgent = True
-                hook.fire("client_urgent_hint_changed", self)
+        # For "never", do nothing.
+        if focus_on_window_activation == "never":
```

Comment:
Isn't it against the whole idea of this PR? You started it after we talked that the urgency should be triggered every time, despite setting of `focus_on_window_activation`. In this version, when the variable is set to `never`, the function will exit and nothing will happen with the urgency - which is basically the same state we have now.